dds/DCPS/transport/tcp/TcpConnection.cpp

```diff
@@ -630,19 +648,17 @@ OpenDDS::DCPS::TcpConnection::passive_reconnect_i()
     this->reconnect_state_ = PASSIVE_WAITING_STATE;
     this->link_->notify(DataLink::DISCONNECTED);
 
-    // It is possible that the passive reconnect is called after the new connection
-    // is accepted and the receive_strategy of this old connection is reset to nil.
-    TcpReceiveStrategy_rch receive_strategy = this->receive_strategy();
-    if (this->receive_strategy()) {
-
-      // Give a copy to reactor.
-      this->passive_reconnect_timer_id_ = receive_strategy->get_reactor()->schedule_timer(this, 0, timeout);
-
-      if (this->passive_reconnect_timer_id_ == -1) {
-        ACE_ERROR_RETURN((LM_ERROR,
-                          ACE_TEXT("(%P|%t) ERROR: TcpConnection::passive_reconnect_i")
-                          ACE_TEXT(", %p.\n"), ACE_TEXT("schedule_timer")),
-                         -1);
+    if (interceptor_ && passive_reconnect_timer_id_ == -1) {
+      ReactorInterceptor::CommandPtr cmd = interceptor_->execute_or_enqueue(new ScheduleTimer(RcHandle<TcpConnection>(this, inc_count()), 0, timeout));
+      guard.release();
+      long result = static_rchandle_cast<ReactorInterceptor::ResultCommand<long> >(cmd)->wait_result();
```

Comment:
I think we could avoid the cast here by providing a static member of `ResultCommand<T>`:
```
  static T wait_result(CommandPtr cmd);
```
To make the calling code: `const long result = ReactorInterceptor::ResultCommand<long>::wait_result(cmd);`